slack-server-integration-common/src/main/java/com/atlassian/plugins/slack/api/client/interceptor/RequestIdInterceptor.java

```diff
@@ -0,0 +1,34 @@
+package com.atlassian.plugins.slack.api.client.interceptor;
+
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+import okio.Buffer;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.UUID;
+
+@Component
+@Slf4j
+public class RequestIdInterceptor implements Interceptor {
+    public static final String REQ_ID_HEADER = "X-Atlassian-Request-Id";
+
+    @Override
+    public Response intercept(final Chain chain) throws IOException {
```

Comment:
New interceptors adds a header with a request identifier, which is then used in other interceptors to track how the requests performed.